src/PoolManager.sol

```diff
@@ -221,12 +221,12 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
 
         // Update the swap amount according to the hook's return, and check that the swap type doesnt change (exact input/output)
         params.amountSpecified += hookDeltaInSpecified;
-        if (exactInput ? params.amountSpecified > 0 : params.amountSpecified < 0) revert SwapTypeChanged();
+        if (exactInput ? params.amountSpecified >= 0 : params.amountSpecified <= 0) revert SwapTypeChanged();
```

Comment:
to allow for NoOp behavior, this needs to allow for `params.amountSpecified = 0`. Currently its reverting if the hook uses all of amountSpecified

```suggestion
        if (exactInput ? params.amountSpecified > 0 : params.amountSpecified < 0) revert SwapTypeChanged();
```

Reply:
have you pulled the latest on the branch? I added this today in commit [4071282](https://github.com/Uniswap/v4-core/pull/482/commits/407128278973f082fd4951a8a38a466c3f2a08a9)

Reply:
i see the downstream NoOp behavior, but for some reason you changed this condition from `< 0` to `<= 0`

so whenever amountSpecified == 0 (after accounting for `hookDeltaInSpecified`), its reverting with `SwapTypeChanged` and never reaching the NoOp condition

Reply:
oh yeah sorry youre right